### ISSUE TITLE
xml syntax highlighting extremely slow

### ISSUE THREAD

**Issue by peskal** [martanne/vis#603]
Editing large xml files with syntax highlighting on is very slow. There is a delay of almost a second each time the cursor is moved. Turning syntax highlighting off makes the cursor usable again. Here is a large xml file to test on:
https://github.com/android/platform_manifest/blob/master/default.xml

Interestingly, I have not seen this issue with other languages.

My ```:help```
```
vis v0.4

 Lua paths used to load C libraries (? will be replaced by filename):

  /usr/lib/lua/5.3/?.so
  /usr/lib/lua/5.3/loadall.so

 Compile time configuration

  Curses support:                 	yes
  Lua support:                    	yes
  Lua LPeg statically built-in:   	no
  TRE based regex support:        	no
  POSIX ACL support:              	no
  SELinux support:                	no
```
I also have ```/usr/lib/lua/5.3/lpeg.so``` on my system.

**Comment by michaelforney** [comment]
I've noticed this as well. Editing XML with syntax highlighting is unusably slow, so I disable it in my `visrc.lua`:

```lua
if win.file.path and win.file.path:find('%.xml$') then
  vis:command('set syntax off')
end
```

I did a bit of investigation, and found that if I comment out `{'equals', equals}` in `lexers/xml.lua`, it performs much better. I suspect https://github.com/martanne/vis/blob/d7e0cbd7795bec43e9e9f11a42a7019f18bf9e18/lua/lexers/xml.lua#L24 might be the culprit.

**Comment by Shugyousha** [comment]
On Tue, Sep 12, 2017 at 10:45 PM, Michael Forney
<notifications@github.com> wrote:
> I've noticed this as well. Editing XML with syntax highlighting is unusably
> slow, so I disable it in my visrc.lua:
>
> if win.file.path and win.file.path:find('%.xml$') then
>   vis:command('set syntax off')
> end

Yes, that is one of the viable workarounds. The other one is to limit
the syntax highlighting search space ("horizon") to a managable value
using https://github.com/martanne/vis/blob/d7e0cbd7795bec43e9e9f11a42a7019f18bf9e18/lua/vis-std.lua#L34
.


Cheers,

Silvan

**Comment by martanne** [comment]
If not fixed, this should at least be improved by 7e9e0a2ca868aaa214fb38a79fe71da34d6e00da. I'm closing this issue for now, feel free to re-open/comment if you still encounter performance problems.
